ios-coding-guidelines.md

```diff
@@ -2,24 +2,39 @@
 
 1. **Avoid Code Duplications.** This way you avoid limitless growing codebase plus, more important, you avoid fixing issues in multiple places. This is one of the most important rules.
 1. **Use the `self.` syntax exclusively in closures and appropriate initializers.** So it will become easier to track down future retain cycles that are caused by missing `[weak self]` in the closure definition. Just leave out `self` if it is not necessary. Also use the `self.` syntax to initially assign constructor parameters to their respective properties if they have the same name.
-1. **Never use the default case in switch statements.** Switch is an extremely useful construct in Swift since the compiler warns about missing cases. If you use the default case you disable this behaviour and loose one of the most powerful features of Swift.
+1. **Dont use the default case if it's possible that the enum is extended in the future.** Switch is an extremely useful construct in Swift since the compiler warns about missing cases. If you use the default case you disable this behaviour and loose one of the most powerful features of Swift. It makes sense however to use the default case if you don't want case-specific behavior for every case, but onle for one or some of them. In this case it doesn't break anything if new cases get added. Also if you switch over plain value types, a default case is harmless.
 1. **Never omit specifying the value of string-typed enum cases.** Imagine you create a string-typed enum and omit the values for every case since Swift infers the value from the case name. Now when you refactor one enum case from somewhere in your app, the inferred value also changes, which is not desired i.e. in JSON keys. You would introduce hard to find errors.
-1. **Create all classes as `final`.** This way some optimisations are enabled for these classes which speeds up compilation.
+1. **Avoid subclasses and finalize your classes** Because super- and subclass logic becomes hard to maintain, try to split up logic into objects instead. By marking your classes final, some optimizations are enabled which among other things speeds up compilation.
```

Comment:
- Avoid subclasses and finalize your classes > Avoid subclasses where possible and finalize your classes
- into objects instead > into separate structs or classes instead
- Add: Structs are final by design. You don't need to mark them final explicitly